data/scripts/actions/others/doors.lua

```diff
@@ -79,9 +79,14 @@ function door.onUse(player, item, fromPosition, target, toPosition, isHotkey)
 			return false
 		end
 		target = tile:getTopVisibleThing()
-		if target.actionid > 0 then
+		if target.actionid > 0 and not table.contains(keys, target.itemid) then
 			if item.actionid == target.actionid then
+				local DOORSTATE = target:getCustomAttribute(ITEM_ATTRIBUTE_DOORSTATE) == DOOR_STATE_UNLOCKED and DOOR_STATE_LOCKED or DOOR_STATE_UNLOCKED
```

Comment:
this is very ugly, please use normal camelCase coding style here

Reply:
Hopefully this is what you meant?

Reply:
yeah, great